File: benchmarks/tsb/bench_explode.ts
import { Series } from "tsb";

const rng = (seed: number) => { let s = seed; return () => { s = (s * 1664525 + 1013904223) & 0xffffffff; return (s >>> 0) / 0xffffffff; }; };
const rand = rng(42);
const data = Array.from({ length: 10_000 }, () => {
  const len = Math.floor(rand() * 5) + 1;
  return Array.from({ length: len }, () => Math.floor(rand() * 100));
});
const s = new Series(data);
for (let i = 0; i < 3; i++) s.explode();
const N = 50;
const t0 = performance.now();
for (let i = 0; i < N; i++) s.explode();
const elapsed = performance.now() - t0;
console.log(JSON.stringify({ function: "explode", mean_ms: elapsed / N, iterations: N, total_ms: elapsed }));

File: benchmarks/pandas/bench_explode.py
import pandas as pd, json, time, numpy as np
rng = np.random.default_rng(42)
# Each row has a list of 1-5 items
data = [[int(x) for x in rng.integers(0, 100, size=rng.integers(1, 6))] for _ in range(10_000)]
s = pd.Series(data)
for _ in range(3): s.explode()
N = 50
t0 = time.perf_counter()
for _ in range(N): s.explode()
elapsed = time.perf_counter() - t0
print(json.dumps({"function": "explode", "mean_ms": elapsed/N*1000, "iterations": N, "total_ms": elapsed*1000}))
